spec/alephant/logger/json_spec.rb

```diff
@@ -1,137 +1,92 @@
 require 'date'
 require 'spec_helper'
 require 'alephant/logger/json'
+require 'alephant/logger/level'
+require_relative 'support/json_shared_examples'
 
 describe Alephant::Logger::JSON do
-  subject { described_class.new(log_path) }
-
   let(:fn)       { -> { 'foo' } }
   let(:log_path) { '/log/path.log' }
-  let(:log_file) { instance_double File }
+  let(:log_output_obj) { instance_double(File) }
+  let(:msg) { :write }
 
   before do
-    allow(File).to receive(:open) { log_file }
-    allow(log_file).to receive :sync=
+    allow(File).to receive(:open) { log_output_obj }
+    allow(log_output_obj).to receive :sync=
   end
 
-  shared_examples 'JSON logging' do
-    let(:log_hash) do
-      { 'foo' => 'bar', 'baz' => 'quux' }
-    end
-
-    it 'writes JSON dump of hash to log with corresponding level key' do
-      allow(Time).to receive(:now).and_return('foobar')
+  logging_levels = Alephant::Logger::LEVELS.map(&:to_s)
+  levels_size = logging_levels.size - 1
 
-      expect(log_file).to receive(:write) do |json_dump|
-        h = { 'timestamp' => 'foobar', 'uuid' => 'n/a', 'level' => level }
-        expect(JSON.parse(json_dump)).to eq h.merge log_hash
-      end
-
-      subject.send(level, log_hash)
-    end
-
-    it 'automatically includes a timestamp' do
-      expect(log_file).to receive(:write) do |json_dump|
-        t = JSON.parse(json_dump)['timestamp']
-        expect { DateTime.parse(t) }.to_not raise_error
-      end
-
-      subject.send(level, log_hash)
-    end
+  %w(debug info warn error).each_with_index do |level, i|
+    describe "##{level}" do
+      let(:level) { level }
+      let(:log_hash) { { 'foo' => 'bar', 'baz' => 'quux' } }
 
-    it 'outputs the timestamp first' do
-      expect(log_file).to receive(:write) do |json_dump|
-        h = JSON.parse(json_dump)
-        expect(h.first[0].to_sym).to be :timestamp
-      end
+      context 'when the desired write level is specified' do
+        subject(:logger) do
+          described_class.new(log_path, level: desired_write_level)
+        end
 
-      subject.send(level, log_hash)
-    end
+        context 'when same as the desired write level' do
```

Comment:
I think we need to add a bit more to the description, as when running specs in documentation format will give us:
"when the desired write level is specified: when same as the desired write level", which will be the same for all of these descriptions inside the context.

You can run the tests in the documentation format:
```sh
bundle exec rspec --format documentation
```

Reply:
I'd add message level in these somewhere just for clarity: "when message level same as the desired write level"